spot-ml/src/main/scala/org/apache/spot/lda/SpotLDAWrapper.scala

```diff
@@ -85,41 +80,67 @@ object SpotLDAWrapper {
         wordDictionary,
         sqlContext)
 
+    val corpusSize = ldaCorpus.count()
+
     docWordCountCache.unpersist()
 
     //Instantiate optimizer based on input
-    val optimizer = new EMLDAOptimizer
+    val ldaOptimizer = ldaOptimizerOption match {
+      case "em" => new EMLDAOptimizer
+      case "online" => new OnlineLDAOptimizer().setOptimizeDocConcentration(true).setMiniBatchFraction({
+        if (corpusSize < 2) 0.75
+        else (0.05 + 1) / corpusSize
+      })
+      case _ => throw new IllegalArgumentException(
+        s"Invalid LDA optimizer $ldaOptimizerOption")
+    }
 
     logger.info(s"Running Spark LDA with params alpha = $ldaAlpha beta = $ldaBeta " +
-      s"Max iterations = $maxIterations Optimizer = em")
-    //Set LDA params from input args
+      s"Max iterations = $maxIterations Optimizer = $ldaOptimizerOption")
 
+    //Set LDA params from input args
     val lda =
       new LDA()
         .setK(topicCount)
         .setMaxIterations(maxIterations)
         .setAlpha(ldaAlpha)
         .setBeta(ldaBeta)
-        .setOptimizer(optimizer)
+        .setOptimizer(ldaOptimizer)
 
     // If caller does not provide seed to lda, ie. ldaSeed is empty, lda is seeded automatically set to hash value of class name
 
     if (ldaSeed.nonEmpty) {
       lda.setSeed(ldaSeed.get)
     }
 
+    val (wordTopicMat, docTopicDist) = ldaOptimizer match {
+      case _: EMLDAOptimizer => {
+         val ldaModel = lda.run(ldaCorpus).asInstanceOf[DistributedLDAModel]//.toLocal
+
+        //Get word topic mix: columns = topic (in no guaranteed order), rows = words (# rows = vocab size)
+        val wordTopicMat: Matrix = ldaModel.topicsMatrix
 
-    //Create LDA model
-    val ldaModel = lda.run(ldaCorpus)
+        //Topic distribution: for each document, return distribution (vector) over topics for that docs
+        val docTopicDist: RDD[(Long, Vector)] = ldaModel.topicDistributions
 
-    //Convert to DistributedLDAModel to expose info about topic distribution
-    val distLDAModel = ldaModel.asInstanceOf[DistributedLDAModel]
+        (wordTopicMat, docTopicDist)
 
-    //Get word topic mix: columns = topic (in no guaranteed order), rows = words (# rows = vocab size)
-    val wordTopicMat: Matrix = distLDAModel.topicsMatrix
+      }
 
-    //Topic distribution: for each document, return distribution (vector) over topics for that docs
-    val docTopicDist: RDD[(Long, Vector)] = distLDAModel.topicDistributions
+      case _: OnlineLDAOptimizer => {
+        val ldaModel = lda.run(ldaCorpus).asInstanceOf[LocalLDAModel]
+
+        //Get word topic mix: columns = topic (in no guaranteed order), rows = words (# rows = vocab size)
+        val wordTopicMat: Matrix = ldaModel.topicsMatrix
+
+        //Topic distribution: for each document, return distribution (vector) over topics for that docs
```

Comment:
minor formatting issue throughout (easy query replace)
white space after // 
eg.
//Topic
should be 
// Topic